theano/tensor/basic.py

```diff
@@ -1427,21 +1427,31 @@ class Argmax(Op):
     nin = 2  # tensor, axis
     nout = 1
     E_axis = 'invalid axis'
-    __props__ = ()
+    __props__ = ('axis',)
```

Comment:
I thinked of another option. You could just implement the prepare_node() method and make it raise a good error if we try to compile an old node. I discussed this IRL with @lamblin and he also agreed with this. What do you think?